pydantic/types.py

```diff
@@ -870,4 +872,60 @@ def __repr__(self) -> str:
             return 'NaiveDatetime'
 
 
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Encoded TYPES ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+class EncoderProtocol(Protocol):
+    @classmethod
+    def decode(cls, data: bytes) -> bytes:
+        ...
+
+    @classmethod
+    def encode(cls, value: bytes) -> bytes:
+        ...
+
+
+class Base64Encoder(EncoderProtocol):
+    @classmethod
+    def decode(cls, data: bytes) -> bytes:
+        return base64.decodebytes(data)
+
+    @classmethod
+    def encode(cls, value: bytes) -> bytes:
+        return base64.encodebytes(value)
+
+
+@_dataclasses.dataclass
```

Comment:
In general, I would use `from ._internal._internal_dataclass import slots_dataclass` here, you _could_ just add `slots=True` but I think it helps convey intent that it's a dataclass used for internal purposes that.

Reply:
Ok. It's worth noting though that other types above use neither `._internal._internal_dataclass` not `slots`.

Reply:
if they can it's probably a good idea to, if it causes any errors it can stay as is